llvm/lib/Analysis/DXILMetadataAnalysis.cpp

```diff
@@ -25,6 +25,14 @@ static ModuleMetadataInfo collectMetadataInfo(Module &M) {
   MMDAI.DXILVersion = TT.getDXILVersion();
   MMDAI.ShaderModelVersion = TT.getOSVersion();
   MMDAI.ShaderStage = TT.getEnvironment();
+  NamedMDNode *ValidatorVerNode = M.getNamedMetadata("dx.valver");
+  if (ValidatorVerNode) {
+    auto *ValVerMD = cast<MDNode>(ValidatorVerNode->getOperand(0));
+    auto *MajorMD = mdconst::extract<ConstantInt>(ValVerMD->getOperand(0));
+    auto *MinorMD = mdconst::extract<ConstantInt>(ValVerMD->getOperand(1));
```

Comment:
Should there be a check if the tuple isn't well formed, or is there a separate place that's already checked?
AFAIK, there's nothing that guarantees someone couldn't form a bad metadata header, which shouldn't be accidentally evaluated.

Reply:
> Should there be a check if the tuple isn't well formed, or is there a separate place that's already checked? AFAIK, there's nothing that guarantees someone couldn't form a bad metadata header, which shouldn't be accidentally evaluated.

These checks are not included here as they are expected to be done during Module construction.